### PR TITLE
docs: Add Authentication section

### PR DESCRIPTION
**Pull request by jonoliver**
This was originally supposed to be a Password Requirement section, but I thought a more general Authentication section would allow us to expand on authentication topics as needed.

I’d love to have some feedback on what I have so far. It feels weird saying “we” for some of the points without a good discussion/critique.

### CODE REVIEW

File: style/authentication/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+Because security is a constantly moving target, we recommend that you [don't roll your own authentication](https://blog.codeship.com/why-you-shouldnt-roll-your-own-authentication/), and instead rely on a battle-tested authentication mechanism. As an example, for Rails apps we use [Devise](https://github.com/plataformatec/devise).
+
+## Password Requirements
+
+We follow the [OWASP Password Guidelines](https://www.owasp.org/index.php/Authentication_Cheat_Sheet#Implement_Proper_Password_Strength_Controls) for strong passwords:
+
+> The password change mechanism should require a minimum level of complexity that makes sense for the application and its user population. For example:
+>
+> Password must meet at least 3 out of the following 4 complexity rules
```
Comment (catheraaine) [inline style/authentication/README.md]: colon after rules

> rules:

File: style/authentication/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+>
+> - at least 1 uppercase character (A-Z)
+> - at least 1 lowercase character (a-z)
+> - at least 1 digit (0-9)
+> - at least 1 special character (punctuation) — do not forget to treat space as special characters too
```
Comment (catheraaine) [inline style/authentication/README.md]: I would rephrase as

> - at least 1 special character (punctuation)—a space counts as a special character

File: style/authentication/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+> - at least 10 characters
+> - at most 128 characters
+> - not more than 2 identical characters in a row (e.g., 111 not allowed)
```
Comment (catheraaine) [inline style/authentication/README.md]: great example!

File: style/authentication/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+We're also a fan of _passphrases over passwords_.
```
Comment (catheraaine) [inline style/authentication/README.md]: TIL

File: style/authentication/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+![Password Strength](./password_strength.png)
```
Comment (catheraaine) [inline style/authentication/README.md]: love this!!

File: style/authentication/README.md (hunk 2 above)
Comment (jonoliver) [inline style/authentication/README.md]: I copied this section from the OWASP guidelines, but I agree that the wording could be improved. I'll update it.

File: style/authentication/README.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
+# Authentication
+
+When building applications that use authentication, security is crucial. To help ensure that we are building secure applications, we follow the [OWASP Authentication General Guidelines](https://www.owasp.org/index.php/Authentication_Cheat_Sheet#Authentication_General_Guidelines).
+
+Because security is a constantly moving target, we recommend that you [don't roll your own authentication](https://blog.codeship.com/why-you-shouldnt-roll-your-own-authentication/), and instead rely on a battle-tested authentication mechanism. As an example, for Rails apps we use [Devise](https://github.com/plataformatec/devise).
+
```
Comment (silverli) [inline style/authentication/README.md]: for Javascript/React things, we've also used [Passport JS](http://passportjs.org/) if you want to add that in as a resource.

File: style/authentication/README.md (hunk 5 above)
Comment (silverli) [reply style/authentication/README.md]: this is amazing. we should explain more PR's with xkcd when possible

**Comment by catheraaine** [comment]
See: #117

**Comment by cromwellryan** [comment]
Lgtm

**Comment by silverli** [comment]
Merged!

https://media.giphy.com/media/wZaoi0FCUFD6U/giphy.gif
